### PR TITLE
add description of xCAT certs and their usage in xCAT SSL communication

### PR DESCRIPTION
**Pull request by immarvin**
http://10.3.5.21/xcat-doc/html/advanced/security/certs.html

### CODE REVIEW

File: docs/source/advanced/security/certs.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
+The Certificates in xCAT
```
Comment (whowutwut) [inline docs/source/advanced/security/certs.rst]: Suggest maybe just  "SSL Certificates"

File: docs/source/advanced/security/certs.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      * file on service node: ``/root/.xcat/ca.pem``
+      * distribution path: 
+        **/etc/xcat/cert/ca.pem (MN)** ===(run ``xcatconfig`` command)===> **/install/postscripts/_xcat/ca.pem (MN)** ===(node provision/updatenode)==> **/xcatpost/_xcat/ca.pem (SN and CN)** ==(run "servicenode" postscript)==> **/root/.xcat/ca.pem (SN)**
+
+   2. xCAT server credential(server-cred.pem): 
+
+      * a concatenation of server private key and certificate(signed with xCAT CA certificate)
+      * generated by ``/opt/xcat/share/xcat/scripts/setup-server-cert.sh`` on xCAT installation;
+      * will be generated (or updated) on xCAT management node when:
+         * install or update xCAT when "/etc/xcat/cert" directory does not exist
```
Comment (whowutwut) [inline docs/source/advanced/security/certs.rst]: Should we make `/etc/xcat/cert` code formatted with two ticks.

File: docs/source/advanced/security/certs.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      * file on service node: ``/etc/xcat/cert/server-cred.pem`` 
+      * distribution path: 
+        **/etc/xcat/cert/server-cred.pem (MN)** ==(run ``xcatserver`` script called by ``servicenode`` postscript)===> **/etc/xcat/cert/server-cred.pem(SN)**
+
+   3. xCAT client credential(client-cred.pem):
+
+      * a concatenation of client private key and certificate (signed with xCAT CA certificate)
+      * generated by ``/opt/xcat/share/xcat/scripts/setup-local-client.sh`` on xCAT installation
+      * will be generated (or updated) on xCAT management node when: 
+         * install or update xCAT when "/root/.xcat/client-key.pem"  does not exist;
```
Comment (whowutwut) [inline docs/source/advanced/security/certs.rst]: Should we make `/root/.xcat/client-key.pem` code formatted with two ticks.

**Comment by daniceexi** [comment]
Generally, the content looks good.

* Could we add kind of page like 101 questions for frequently asked questions and then link the answer to corresponding pages. Otherwise it's hard for a user to get to this page when they want it.
* There was a question about how to set up the ssl version for communication between xcat_client and xcat_server. Do we want to cover it somewhere?

File: docs/source/advanced/security/certs.rst (hunk 1 above)
Comment (immarvin) [inline docs/source/advanced/security/certs.rst]: updated

File: docs/source/advanced/security/certs.rst (hunk 2 above)
Comment (immarvin) [inline docs/source/advanced/security/certs.rst]: updated

File: docs/source/advanced/security/certs.rst (hunk 3 above)
Comment (immarvin) [inline docs/source/advanced/security/certs.rst]: updated

**Comment by whowutwut** [comment]
Let's merge this documentation in, we can add to it once it's out there in the doc.  Thanks @immarvin

**Pull request merged**